### PR TITLE
Use the proper type mapping for the result of AVG

### LINKED ISSUE #26378: Cosmos: Average truncates results
This matches behavior in Cosmos SQL, but Smit suggested I still file this.

```C#
public class Blog
{
    public Guid Id { get; set; }
    public int Value { get; set; }
}

public class SomeDbContext : DbContext
{
    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        => optionsBuilder
            .UseCosmos(
                "https://localhost:8081",
                "C2y6yDjf5/R+ob0N8A7Cgv30VRDJIWEHLM+4QDU5DE2nQ9nDuVTqobD4b8mGGyPMbIZnqyMsEcaGQy67XIw/Jw==",
                "Daily")
            .LogTo(Console.WriteLine, LogLevel.Information)
            .EnableSensitiveDataLogging();

    public virtual DbSet<Blog> Blogs { get; set; }
}

public class Program
{
    public static void Main()
    {
        using (var context = new SomeDbContext())
        {
            context.Database.EnsureDeleted();
            context.Database.EnsureCreated();

            context.Add(new Blog { Value = 2 });
            context.Add(new Blog { Value = 3 });
            context.SaveChanges();

            Console.WriteLine($"Average value = {context.Blogs.Select(e => e.Value).Average()}.");
        }
    }
}
```

```
info: 10/16/2021 12:16:48.167 CosmosEventId.ExecutedReadNext[30102] (Microsoft.EntityFrameworkCore.Database.Command)
      Executed ReadNext (217.5037 ms, 3.1 RU) ActivityId='(null)', Container='SomeDbContext', Partition='(null)', Parameters=[]
      SELECT AVG(c["Value"]) AS c
      FROM root c
      WHERE (c["Discriminator"] = "Blog")
Average value = 2.
```

### PR DESCRIPTION
**Pull request by ChrisJollyAU**
The translation of `Average` in Cosmos uses the same type mapping of its input for its return.

While this is fine if its input is a float, double, decimal, if you have an int or long as input it sets the return type as the same, thus causing a truncation when the results are read.

This fixes it by using the `resultType` passed in (the result type of the C# Average method - either float or double)

Fixes #26378

### CODE REVIEW

File: src/EFCore.Cosmos/Query/Internal/CosmosQueryableMethodTranslatingExpressionVisitor.cs [hunk 1]
```diff
@@ -459,7 +459,7 @@ private ShapedQueryExpression CreateShapedQueryExpression(SelectExpression selec
         }
 
         var projection = (SqlExpression)selectExpression.GetMappedProjection(new ProjectionMember());
-        projection = _sqlExpressionFactory.Function("AVG", new[] { projection }, projection.Type, projection.TypeMapping);
+        projection = _sqlExpressionFactory.Function("AVG", new[] { projection }, resultType, _typeMappingSource.FindMapping(resultType) ?? projection.TypeMapping);
```
Comment (roji) [inline src/EFCore.Cosmos/Query/Internal/CosmosQueryableMethodTranslatingExpressionVisitor.cs:462]: This is mostly a nit, but it's probably slightly safer/clearer to simply not translate if we can't find the type mapping? What do you think?

**Review by roji — APPROVED** [review]
@ChrisJollyAU thanks for fixing this!

I'm approving, but see my nit comment - will delay merging for that for now.

File: src/EFCore.Cosmos/Query/Internal/CosmosQueryableMethodTranslatingExpressionVisitor.cs (hunk 1 above)
Comment (ChrisJollyAU) [reply src/EFCore.Cosmos/Query/Internal/CosmosQueryableMethodTranslatingExpressionVisitor.cs:462]: Was kind of undecided. At one point was just going to leave it at `_typeMappingSource.FindMapping(resultType)` without the fallback. Probably unlikely to be needed as `resultType` is going to be either `double` or `float` based on the C# method definition. And that is almost guaranteed to be found

Comment (roji) [reply src/EFCore.Cosmos/Query/Internal/CosmosQueryableMethodTranslatingExpressionVisitor.cs:462]: I agree that this is unlikely to actually matter, but I do have a slight preference to remove the fallback, unless you have an objection; but am fine to merge as-is if you prefer.

Comment (ChrisJollyAU) [reply src/EFCore.Cosmos/Query/Internal/CosmosQueryableMethodTranslatingExpressionVisitor.cs:462]: No real feelings either way. Will remove fallback

Comment (ChrisJollyAU) [reply src/EFCore.Cosmos/Query/Internal/CosmosQueryableMethodTranslatingExpressionVisitor.cs]: Changes made

**Pull request merged**